zephyr/boot_test.c

```diff
@@ -4,9 +4,19 @@
  */
 
 #include <zephyr/logging/log.h>
+#include <sof\boot_test.h>
 
 #include <zephyr/ztest.h>
 
 LOG_MODULE_REGISTER(sof_boot_test, LOG_LEVEL_DBG);
 
 ZTEST_SUITE(sof_boot, NULL, NULL, NULL, NULL, NULL);
+
+#if CONFIG_SOF_BOOT_TEST
+void sys_run_boot_tests(void)
+{
+	TEST_RUN_ONCE(ztest_run_test_suite, sof_boot);
+}
+SYS_INIT(sys_run_boot_tests, APPLICATION, 99);
+
+#endif
```

Comment:
```
SYS_INIT(sys_run_boot_tests, APPLICATION, 99);
```
99 here guarantees execution after everything else. Works in my testing.

Reply:
99 doesn't *guarantee* that - others might also use 99 (even if it's the maximum accepted number there). And it's only the ordering of init calls, whereas in my testing something was missing, that was done after all the init calls. But if it's working now - that's good, something must've changed since I was testing it